nilearn/_utils/niimg_conversions.py

```diff
@@ -102,7 +102,23 @@ def _iter_check_niimg(niimgs, ensure_ndim=None, atleast_4d=False,
         Data type toward which the data should be converted. If "auto", the
         data will be converted to int32 if dtype is discrete and float32 if it
         is continuous.
+
+    See also
+    --------
+        check_niimg, check_niimg_3d, check_niimg_4d
     """
+    # Verify the input list is not empty, otherwise raise an error.
+    if isinstance(niimgs, list) and len(niimgs) == 0:
+        raise ValueError("Input niimgs list is empty.")
+
+    # If niimgs is a string, use glob to expand it to the matching filenames.
+    if isinstance(niimgs, _basestring):
+        niimgs_list = glob.glob(os.path.expanduser(niimgs))
+        # Raise an error in case the niimgs list is empty.
+        if len(niimgs_list) == 0:
+            raise ValueError("No files matching path: %s" % niimgs)
```

Comment:
There's a small difference in the error message which return the input filename.